public/react/Theme/Theme.tsx

```diff
@@ -0,0 +1,100 @@
+import type { FC } from "react";
+
+import React, { useEffect, useState } from "react";
+import SidebarButton from "../components/SidebarButton";
+
+import ArrowLeftCircleIcon from "@atlaskit/icon/glyph/arrow-left-circle";
+import Heading from "@atlaskit/heading";
+import Lozenge from "@atlaskit/lozenge";
+import { getTextColorUsingAPCA } from "../../utils/color";
+import { useTheme } from "../services/theme/useTheme";
+import { useSaveTheme } from "../services/theme/useSaveTheme";
+import { applyThemeToCssVars, defaultTheme, Theme } from "../../jira/theme";
+import { useDebounce } from "../hooks/useDebounce";
+import Spinner from "@atlaskit/spinner";
+import Button from "@atlaskit/button/new";
+
+interface ThemeProps {
+  onBackButtonClicked: () => void;
+}
+
+const Theme: FC<ThemeProps> = ({ onBackButtonClicked }) => {
+  const theme = useTheme();
+  const { save, isPending } = useSaveTheme();
+
+  const [localTheme, setLocalTheme] = useState(theme);
+
+  const updateLocalTheme = (newLocalTheme: Theme) => {
+    applyThemeToCssVars(newLocalTheme);
+    setLocalTheme(newLocalTheme);
+  };
+
+  const debouncedTheme = useDebounce(localTheme, 500);
+
+  useEffect(() => {
+    if (JSON.stringify(theme) === JSON.stringify(debouncedTheme)) {
```

Comment:
Theme is an object right? Is it possible that you need to sort the keys for this to always work?  

```
JSON.stringify({a: 1, b:2}) !== JSON.stringify({b:2, a: 1})
```

Reply:
Yeah, this is primarily to protect agains firing off a save on load... I could do some sorting too... what do you think